cartridges/int_klaviyo_sfra/cartridge/controllers/Order.js

```diff
@@ -29,6 +29,23 @@ server.append('Confirm', function (req, res, next) {
                     dataObj = orderConfirmationData.getData(currentOrder, exchangeID);
                     klaviyoUtils.trackEvent(exchangeID, dataObj, klaviyoUtils.EVENT_NAMES.orderConfirmation, currentOrder.customerEmail);
                 }
+
+                if ('KLEmailSubscribe' in session.custom || 'KLSmsSubscribe' in session.custom) {
+                    var email = session.custom.KLEmailSubscribe ? currentOrder.customerEmail : false;
+                    var phone = session.custom.KLSmsSubscribe ? currentOrder.defaultShipment.shippingAddress.phone : false;
+                    var e164PhoneRegex = new RegExp(/^\+[1-9]\d{1,14}$/);
+                    if (phone) {
+                        // NOTE: Klaviyo only accepts phone numbers that include + and the country code at the start (ex for US: +16465551212)
+                        // in order to successfully get users subscribed to SMS list you must collect the country code in your order phone number field!
+                        phone = '+' + phone.replace(/[^a-z0-9]/gi, '');
+                        if (!e164PhoneRegex.test(phone)) {
+                            phone = false;
+                        }
+                    }
+                    if (email || phone) {
+                        klaviyoUtils.subscribeUser(email, phone);
+                    }
+                }
```

Comment:
```suggestion
                if ('KLEmailSubscribe' in session.custom || 'KLSmsSubscribe' in session.custom) {
                    var email = currentOrder.customerEmail;
                    var phone = currentOrder.defaultShipment.shippingAddress.phone;
                    var e164PhoneRegex = new RegExp(/^\+[1-9]\d{1,14}$/);
                    if (phone) {
                        // NOTE: Klaviyo only accepts phone numbers that include + and the country code at the start (ex for US: +16465551212)
                        // in order to successfully get users subscribed to SMS list you must collect the country code in your order phone number field!
                        phone = '+' + phone.replace(/[^a-z0-9]/gi, '');
                        if (!e164PhoneRegex.test(phone)) {
                            phone = null;
                        }
                    }
                    if (session.custom.KLEmailSubscribe || session.custom.KLSmsSubscribe) {
                        klaviyoUtils.subscribeUser(email, phone,session.custom.KLEmailSubscribe,session.custom.KLSmsSubscribe);
                    }
                }
```

Reply:
@cykolln - Thanks for the feedback. I'm not sure I follow the suggested changes in this file. These suggested changes essentially do the same thing that's already in place - and they pass session values to the subscribeUser function when those are globally available. 

For example, the turnery check for email determines if the session has KLEmailSubscribe in it - and if it does, then the customer email is stored in that variable. The check before subscribeUser is run confirms whether an email or phone is present. 

As for passing the `session.custom.KLEmailSubscribe`, that value is stored on the session when a user checks one of the boxes. It occurs via an ajax call in Klaviyo.js. When the session is updated in Klaviyo.js after that ajax call, the state is maintained and that session data is available throughout the site. So we wouldn't need to pass that value into the function because that value is actually available in utils.js within subscribeUser if we needed that.

Reply:
Yes lets keep the existing approach for calling `subscribeUser` (no need to pass the global vars, that's my bad) but the main change here is we will want to pass both email and phone number if either session var is set. The only time where we should not pass both would be if the validation regex for phone number fails. 

Our cases are: 

*Valid phone number and email*
* `session.custom.KLEmailSubscribe` : make API request to Klaviyo to the email `channel`, including both email and phone number. 
* `session.custom.KLSmsSubscribe` : make API request to Klaviyo to the email `sms`, including both email and phone number. 

*email and invalid phone number*
* `session.custom.KLEmailSubscribe`: Make API request to Klaviyo to the email `channel`, including only email. You can leave phone number out, or our endpoint also accepts `null`. Either works.
* `session.custom.KLSmsSubscribe`: log and don't attempt to make the subscribe request.

Reply:
Just updated the suggested change above to reflect this convo - the other change in subscribeUser would essentially check the session vars and make the corresponding requests following the cases above. 

Making an assumption that SFCC handles email validation before an order can be successfully submitted which is why we are only doing additional validation for phone number given the country code required by the klaviyo API.